flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/NettyShuffleEnvironmentConfiguration.java

```diff
@@ -479,6 +491,7 @@ public int hashCode() {
 		result = 31 * result + floatingNetworkBuffersPerGate;
 		result = 31 * result + (isCreditBased ? 1 : 0);
 		result = 31 * result + (nettyConfig != null ? nettyConfig.hashCode() : 0);
+		result = 31 * result + tempDirs.hashCode();
```

Comment:
not sure why we actually need `hashCode/equals` for `NettyShuffleEnvironmentConfiguration`,
but we should use `Arrays.hashCode()` and `Arrays.equals()` for arrays here and in `equals`.

Reply:
I also think `hashCode/equals` are not needed in this class. Or  we do not add the `tempDirs` for these methods which might be removed future in separate ticket. Or  add it as you suggested. Which one do you prefer?

Reply:
True we can consider removal of these methods in a separate issue (we should make sure they are not needed) but then I would prefer to keep them consistent in this PR as long as they exist and add `tempDirs` using the proper array methods.